### ISSUE TITLE
Split SDK into v1.0 and beta (Consider splitting up the SDK into different editions)

### ISSUE THREAD

**Issue by sassdawe** [microsoftgraph/msgraph-sdk-powershell#692]
The current footprint of the SDK is huge ~1 GB as of v1.5.

Could you consider splitting the SDK up into a GA and a Beta version?

In my experience the bigger part of your target audience will never build any production code based on the Beta Microsoft Graph API.

Splitting the SDK up into two: `Microsoft.Graph` and `Microsoft.Graph.Beta`, would decrease the size, and speed up all kinds of operations for example:
- Installing the module
- Updating the module
- Importing the module
- Calling Get-Help on a command
- etc. 

Thanks for considering it!

PS: because most of the module is generated, keeping two versions updated isn't that of a big task(?). 
[AB#9835](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/9835)

**Comment by darrelmiller** [comment]
This is an active conversation and a proposal that we are seriously considering.  If anyone has any reasons for us not to do the split, now would be the time to share them.

**Comment by JustinGrote** [comment]
I guess my only consideration would be the ramifications for converting from beta to GA when developing a module against beta APIs. Today it's just a matter of changing the initialization structure, what would it look like here? Would the cmdlets become Get-MgBetaItem or would they be the same names that would clobber each other?

I would lean towards same cmdlet name so the script doesn't have to be completely refactored. This is assuming that mixing GA and beta modules would not be permitted, it would have to be "all-or-nothing" in terms of which dependency tree you use.

**Comment by darrelmiller** [comment]
We were considering Get-MgBeta* because some people would want to have both modules installed.  The assumption being that switching an existing script to the new module would be a simple search and replace.

**Comment by JustinGrote** [comment]
@darrelmiller I suppose it's not really like a switch between dev and test because once something goes "GA" it is a one-time replace, so I guess that makes more sense.

**Comment by maisarissi** [comment]
Hello everyone and thank you so much for your feedback.
We did some work recently to decrease the size of the PowerShell SDK. The current version (1.9.2) has ~660MB compared to previous versions (1.8.0 or older) hitting ~1GB.

Even though, we have decided to split the SDK into v1 and beta along with other improvements for the next PowerShell SDK v2 version.

Really appreciate your engagement and keep the feedback coming.
